### PR TITLE
Add name and endpoint check for p2p-preheat

### LINKED ISSUE #12682: Chinese UI translation
i noticed some discrepancies in Harbor Chinese UI,  2.0+, please help check and make some adjustments. THanks.

Chinese UI:
1. main page, bottom left corner:
Harbor Api V2.0  -> Harbor API V2.0

2. 开发人员 -> 开发者

if screenshots are needed, please let me know. Thanks.

### PR DESCRIPTION
**Pull request by AllForNothing**
1. Add name and endpoint checking when adding
2. Add server paging for p2p task list
3. Fixes #12682

Temp Env: https://10.185.12.35:4200
Signed-off-by: AllForNothing <sshijun@vmware.com>

### CODE REVIEW

File: src/portal/src/app/project/p2p-provider/task-list/task-list.component.html [hunk 1]
```diff
@@ -76,25 +76,34 @@ <h2 class="custom-h2 h2-style">{{executionId}}</h2>
...
+            <select id="selectKey" (change)="selectFilterKey($event)">
+              <option value="id">{{"REPLICATION.TASK_ID" | translate | lowercase}}</option>
+              <option value="execution_id">{{"P2P_PROVIDER.ID" | translate | lowercase}}</option>
+              <option value="status">{{"REPLICATION.STATUS" | translate | lowercase}}</option>
+            </select>
+          </div>
+          <hbr-filter (filterEvt)="doFilter($event)" [currentValue]="searchString" id="filter-executions" [withDivider]="true"
+                      (openFlag)="openFilter($event)" filterPlaceholder='{{"REPLICATION.FILTER_PLACEHOLDER" | translate}}'></hbr-filter>
+          <span class="refresh-btn">
+                                 <clr-icon shape="refresh" (click)="refreshTasks()" [hidden]="loading"></clr-icon>
```
Comment (selina-cui) [inline src/portal/src/app/project/p2p-provider/task-list/task-list.component.html:92]: The refresh icon  refreshes policy table not task table.

File: src/portal/src/app/project/p2p-provider/task-list/task-list.component.html [hunk 2]
```diff
@@ -76,25 +76,34 @@ <h2 class="custom-h2 h2-style">{{executionId}}</h2>
...
-      <clr-dg-column [clrDgField]="'id'">{{'REPLICATION.TASK_ID'| translate}}</clr-dg-column>
-      <clr-dg-column [clrDgField]="'execution_id'">{{'P2P_PROVIDER.ID' | translate}}</clr-dg-column>
-      <clr-dg-column [clrDgField]="'status'">{{'REPLICATION.STATUS' | translate}}</clr-dg-column>
+      <clr-dg-action-bar>
+        <div class="row flex-end">
+          <div class="select filter-tag clr-select-wrapper" [hidden]="!isOpenFilterTag">
+            <select id="selectKey" (change)="selectFilterKey($event)">
+              <option value="id">{{"REPLICATION.TASK_ID" | translate | lowercase}}</option>
+              <option value="execution_id">{{"P2P_PROVIDER.ID" | translate | lowercase}}</option>
+              <option value="status">{{"REPLICATION.STATUS" | translate | lowercase}}</option>
```
Comment (selina-cui) [inline src/portal/src/app/project/p2p-provider/task-list/task-list.component.html:86]: status filter doesn't work.

Comment (AllForNothing) [reply src/portal/src/app/project/p2p-provider/task-list/task-list.component.html:86]: It a back-end bug, will be fixed later

File: src/portal/src/app/project/p2p-provider/task-list/task-list.component.html (hunk 1 above)
Comment (AllForNothing) [reply src/portal/src/app/project/p2p-provider/task-list/task-list.component.html:92]: Fixed.

**Pull request merged**